modules/core/src/test/scala/org/tessellation/infrastructure/rewards/RewardsSuite.scala

```diff
@@ -2,141 +2,109 @@ package org.tessellation.infrastructure.rewards
-        )
-      )
-    )
-  )
-
-  test("epoch 1 snapshot reward is valid") { res =>
-    implicit val (_, sp, rand) = res
-    val regular = Regular.make
-
-    for {
-      rewards <- Rewards.make[IO](config, softStaking, dtm, stardust, regular).pure[F]
-      txs <- rewards
-        .calculateRewards(
-          EpochProgress(1L),
-          facilitators
-        )
-      sum = txs.toList.map(_.amount.value.toLong).sum
+  val lowerBound: NonNegLong = EpochProgress.MinValue.value
+  val upperBound: NonNegLong = config.rewardsPerEpoch.keySet.max.value
+  val lowerBoundNoMinting: NonNegLong = NonNegLong.unsafeFrom(upperBound.value + 1)
+  val special: Seq[NonNegLong] =
+    config.rewardsPerEpoch.keys.flatMap(epochEnd => Seq(epochEnd.value, NonNegLong.unsafeFrom(epochEnd.value + 1L))).toSeq
 
-    } yield expect(sum == 65843621389L)
-  }
+  val meaningfulEpochProgressGen: Gen[EpochProgress] =
+    chooseNumRefined(lowerBound, upperBound, special: _*).map(EpochProgress(_))
 
-  test("epoch 2 snapshot reward is valid") { res =>
-    implicit val (_, sp, rand) = res
-    val regular = Regular.make
+  val overflowEpochProgressGen: Gen[EpochProgress] =
+    chooseNumRefined(lowerBoundNoMinting, NonNegLong.MaxValue).map(EpochProgress(_))
 
-    for {
-      rewards <- Rewards.make[IO](config, softStaking, dtm, stardust, regular).pure[F]
-      txs <- rewards
-        .calculateRewards(
-          EpochProgress(1296001L),
-          facilitators
-        )
-      sum = txs.toList.map(_.amount.value.toLong).sum
+  def facilitatorsGen(implicit genIdFn: GenIdFn): Gen[NonEmptySet[Id]] = Gen
+    .nonEmptyListOf(
+      Gen.delay(genIdFn())
+    )
+    .map(_.toNel.get.toNes)
 
-    } yield expect(sum == 32921810694L)
-  }
+  test("generated reward transactions sum up to the total snapshot reward") { res =>
+    implicit val (_, sp, rand, makeIdFn) = res
 
-  test("epoch 3 snapshot reward is valid") { res =>
-    implicit val (_, sp, rand) = res
+    val softStaking = SoftStaking.make(config.softStaking)
+    val dtm = DTM.make(config.dtm)
+    val stardust = StardustCollective.make(config.stardust)
     val regular = Regular.make
 
-    for {
-      rewards <- Rewards.make[IO](config, softStaking, dtm, stardust, regular).pure[F]
-      txs <- rewards
-        .calculateRewards(
-          EpochProgress(1296001L * 2),
-          facilitators
-        )
-      sum = txs.toList.map(_.amount.value.toLong).sum
-
-    } yield expect(sum == 16460905347L)
+    val gen = for {
+      epochProgress <- meaningfulEpochProgressGen
+      facilitators <- facilitatorsGen
+    } yield (epochProgress, facilitators)
+
+    forall(gen) {
+      case (epochProgress, facilitators) =>
+        for {
+          rewards <- Rewards.make[IO](config.rewardsPerEpoch, softStaking, dtm, stardust, regular).pure[F]
+          txs <- rewards.calculateRewards(epochProgress, facilitators)
```

Comment:
I was thinking about extracting an interface like `rewards.distributeAmount(amount: Amount, facilitators: Facilitators)` but it seemed to me like making some internal interfaces public just to make it available for the test suite.

Of course Scala has this fancy mechanism: `private[rewards]` but we cannot use that properly since we use the `make` factory pattern (whatever it's called).